wgpu-core/src/device/mod.rs

```diff
@@ -1913,6 +1879,8 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                 (spv, module)
             }
             pipeline::ShaderModuleSource::Wgsl(code) => {
+                // TODO: refactor the corresponding Naga error to be owned, and then
+                // display it instead of unwrapping
                 let module = naga::front::wgsl::parse_str(&code).unwrap();
```

Comment:
I've tried to use `?` to return this error, but I got a lifetime error. The issue seems to be that `code` is local to this function (yet I think it should work, since the returned lifetime is the same as the shader source's lifetime) :confused:

Reply:
I think we need an explicit lifetime:
```rust
    pub fn device_create_shader_module<'a, B: GfxBackend>(
        &self,
        device_id: id::DeviceId,
        source: pipeline::ShaderModuleSource<'a>,
        id_in: Input<G, id::ShaderModuleId>,
    ) -> Result<id::ShaderModuleId, CreateShaderModuleError<'a>> {
```